Bug/tidyselect not working

Moohan opened the pull request:
This partially resolves #84 I'm pretty sure what I have is an improvement (with no regressions) but it's not a complete fix.

I don't have time to work on this anymore and to be honest, all the non-standard evaluation stuff is outside my comfort zone! It might be worth getting in touch with Nic - nicolaos.christofidis@phs.scot - as he did the work on https://github.com/Public-Health-Scotland/phslookups/pull/9 which is a simpler version of this.

I added tests which you'll probably want to expand, the only one that doesn't work is:
```r
read_slf_episode(
      "1920",
      col_select = c("year", dplyr::contains("dd")),
      recids = "DD"
    )
```
i.e. when a user uses tidyselect helpers *and* uses the filter but (probably) didn't include the required filter variable. In the above example, it's filtering by `recid` but `recid` isn't in `col_select`.

Moohan committed:
Allow using tidyselect helpers with `col_select`

Moohan committed:
Add some tests for tidyselect helpers

Moohan committed:
Update documentation

Jennit07 commented:
Thanks @Moohan appreciate your help with this. Zihao is going to pick this up and look into developing it further.

lizihao-anu committed:
recid and partnership filter
allow for recid and partnership filter when they are not specified to select in columns

lizihao-anu committed:
Style package

lizihao-anu commented:
Honestly, I could not find a way to make tidy selection work outside a *selecting* function. The error looks like below.

```
Error:
! `ends_with()` must be used within a *selecting* function.
ℹ See ?tidyselect::faq-selection-context for details.
```

The workaround is to read_parquet once more for recid or partnerships. Given the super-fast speed of parquet, although it doubles the `read_slf` execution time, it does not really matter. The benchmark shows that the running time increases from 250ms-ish to 500ms-ish.

Moohan commented:
Nice work, I think this solution is fine 👍 I can't see a way to clean up the code anymore than you have it already, and the final case is probably pretty niche (filtering without asking for the relevant filter variable), so probably fine if it's slightly slower in that case.

lizihao-anu committed:
update tests
